app-frontend/src/containers/ProfilePage/LinkCardContainer.tsx

```diff
@@ -60,31 +60,18 @@ const LinkCardContainer:FC<LinkProps> = function ({ link, error }) {
     }
   };
 
-  const getChangedLink = useCallback(async (): Promise<void> => {
-    try {
-      const fetched: RequestPromise = await request(`http://localhost:5000/api/link/${linkId}`, 'GET', null, {
-        Authorization: `Bearer ${auth.token}`,
-      });
-
-      setLinkInfo(fetched.link);
-    } catch (e: any) {
-      setErrorMessage(e.message);
-    }
-  }, [auth.token, linkId, request]);
-
   const confirmChanges = async (): Promise<void> => {
     try {
       setUpload('loading...');
-      const data: RequestPromise = await request('http://localhost:5000/api/link/edit', 'POST', { code: link.code, tags: tagsArray, description }, {
-        Authorization: `Bearer ${auth.token}`,
-      });
-
-      await getChangedLink();
-
-      setEditMessage(data.message);
+      dispatch(editLinkData({
+        code: link.code, tags: tagsArray, description, token: data.token,
+      }));
+      setEditMessage('link has been updated successfully!');
```

Comment:
You should show this message from sagas right after a successful response from the server.